.changes/v3.9.0/1004-improvements.md

```diff
@@ -0,0 +1,4 @@
+* Resources `vcd_vapp_network` and `vcd_vapp_org_network` add convenience flag
+  `reboot_vapp_on_destroy`. When enabled, it will to power off parent vApp (and power back on after
+  if it was before) during vApp network removal. This improves workflows with VCD 10.4.1+ which
+  returns an error when removing vApp networks from powered on vApps [GH-1004]
```

Comment:
I want to open up one more thought before we close this PR. Imagine a vApp with multiple networks. Each network will have to have this flag for `terraform destroy` to work. That's a lot of additional lines in the HCL to manage.

So, I was thinking whether this argument really belongs to the networks? Or is it actually a parameter of vApp itself? In such case, vApp could get a single argument `reboot_vapp_on_network_removal`.

Is there a use case where in a single vApp one network has the new argument set, but another one - doesn't?

CC: @dataclouder @adambarreiro @adezxc

Reply:
This cannot be a parameter of vApp, because we can't access this value during vApp network removals and this happens earlier in the workflow